hw_diag/utilities/quectel.py

```diff
@@ -73,28 +82,35 @@ def find_eg25g_modem() -> Modem:
 @retry(Exception, tries=3, delay=5, backoff=2, logger=logging)
 def download_and_extract(url: str, file_path: str, file_hash: str) -> None:
     download_with_resume(url, file_path, file_hash)
-    untar_cmd = ['tar', '-xzf', file_path, '-C', FW_STORE_PATH]
-    tar_output = ''
+    untar_cmd = ["tar", "-xzf", file_path, "-C", FW_STORE_PATH]
+    tar_output = ""
     try:
         tar_output = subprocess.check_output(untar_cmd, stderr=subprocess.STDOUT)
     except Exception as e:
-        logging.error(f'Failed to untar {file_path}')
+        logging.error(f"Failed to untar {file_path}")
         raise e
     finally:
-        logging.debug(f'{untar_cmd} output: {tar_output}')
+        logging.debug(f"{untar_cmd} output: {tar_output}")
```

Comment:
I would move this into the try, its not a very helpful command if the untar fails. That would also allow you to get rid of setting tar_output to empty string

Reply:
I think it is better if we have tar commands output logged in both cases success or failure.

Reply:
Won't tar_output always be empty string if `check_output` fails since a CalledProcessError will be raised.

Reply:
You sure about that? If `check_output` raises an exception, it can't also return anything can it? From the [docs](https://docs.python.org/3/library/subprocess.html#subprocess.check_output):

> If the return code was non-zero it raises a [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError). The [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) object will have the return code in the [returncode](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError.returncode) attribute and any [output](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError.output) in the output attribute.

Reply:
Done